File: Testing-001/MyApplication/Person.cs
using System;
using System.Collections.Generic;
using System.Threading;

namespace MyApplication.dependencies
{
    public class Person
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="Person"/> class.
        /// </summary>
        /// <param name="values">The field values used for a person.</param>
        public Person(Dictionary<string, dynamic> values)
        {
            Name = values["Name"];
            Occupation = values["Occupation"];
            DOB = values["DOB"];
        }

        public string Name { get; set; }
        public string Occupation { get; set; }
        public DateTime DOB { get; set; }

        /// <summary>
        /// Determines whether the p is valid.
        /// </summary>
        /// <returns></returns>
        public bool IsValid()
        {
            // Simulate some complex processing!
            Thread.Sleep(400);

            // Check that the name needs to contain an even number of letters
            bool result = (Name.Length & 0x1) == 0;
            result &= (Occupation.Length & 0x3) == 0;
            result &= DOB.Day == (Name.Length);
            result &= DOB.Month == (Occupation.Length);

            return result;
        }
    }
}

File: Testing-001-Java/src/main/java/com/tomkuhn/unnittest/workshop/Person.java
package com.tomkuhn.unnittest.workshop;

import java.time.LocalDate;

public class Person {

    private String name;
    private String occupation;
    private LocalDate dob;

    public Person(String name, String occupation, LocalDate dob) {
        this.name = name;
        this.occupation = occupation;
        this.dob = dob;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getOccupation() {
        return occupation;
    }

    public void setOccupation(String occupation) {
        this.occupation = occupation;
    }

    public LocalDate getDob() {
        return dob;
    }

    public void setDob(LocalDate dob) {
        this.dob = dob;
    }

    public boolean isValid()
    {
        try {
            // Simulate some complex processing!
            Thread.sleep(400);
        } catch (InterruptedException e) {
            // Do nothing!
        }

        // Check that the name needs to contain an even number of letters
        boolean result = (name.length() & 1) == 0;

        // Check that the occupation length is a multiple of 3
        result &= (occupation.length() & 3) == 0;

        // Business rules around dob being tied to the name and occupation
        result &= dob.getDayOfMonth() == (name.length());
        result &= dob.getMonthValue() == (occupation.length());

        return result;
    }
}
